File: backend/src/main/java/com/java/NBE4_5_1_7/domain/openAI/dto/ChatResponse.java
package com.java.NBE4_5_1_7.domain.openAI.dto;

import com.java.NBE4_5_1_7.domain.openAI.Message;

import java.util.List;

public class ChatResponse {
    private List<Choice> choices;

    // getters & setters
    public List<Choice> getChoices() {
        return choices;
    }
    public void setChoices(List<Choice> choices) {
        this.choices = choices;
    }

    public static class Choice {
        private Message message;

        // getters & setters
        public Message getMessage() {
            return message;
        }
        public void setMessage(Message message) {
            this.message = message;
        }
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/openAI/dto/ChatResponse.kt
package com.java.NBE4_5_3_7.domain.openAI.dto

import com.java.NBE4_5_3_7.domain.openAI.Message
import com.fasterxml.jackson.annotation.JsonProperty

class ChatResponse {
    @JsonProperty("choices")
    var choices: List<Choice>? = null

    class Choice {
        @JsonProperty("message")
        var message: Message? = null
    }
}